src/jit/lower.cpp

```diff
@@ -462,6 +462,17 @@ void Lowering::DecomposeNode(GenTreePtr* pTree, Compiler::fgWalkData* data)
             hiOp2->gtNext      = hiResult;
             hiResult->gtPrev   = hiOp2;
 
+            if (oper == GT_ADD || oper == GT_SUB)
+            {
+                if (loResult->gtOverflow())
+                {
+                    hiResult->gtFlags |= GTF_OVERFLOW;
+                    if (loResult->gtFlags & GTF_UNSIGNED)
+                    {
+                        hiResult->gtFlags |= GTF_UNSIGNED;
+                    }
+                }
```

Comment:
should we unconditionally propagate GTF_UNSIGNED (not just for add/sub, or only for overflow-possible nodes), e.g.
    hiResult->gtFlags |= (loResult->gtFlags & GTF_UNSIGNED);
?